src/libraries/TypesLib.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "./BLS.sol";
+
+library TypesLib {
+    // Signature request struct for signature request type
+    struct SignatureRequest {
+        bytes message; // plaintext message to hash and sign
+        bytes messageHash; // hashed message to sign
+        bytes condition; // optional condition, length can be zero for immediate message signing
+        string schemeID; // signature scheme id, e.g., "BN254", "BLS12-381", "TESS"
+        address callback; // the requester address to call back. Must implement ISignatureReceiver interface to support the required callback
+        bytes signature;
```

Comment:
minor, but worth commenting these as it's a place downstream consumers are likely to end up if they're confused